File: Examples/Conversations/Obj-C/ConversationsExample/ViewController.m
//
//  ViewController.m
//  ConversationsExample
//
//  Copyright © 2016 Bazaarvoice. All rights reserved.
//

#import "ViewController.h"
#import "MyReviewTableViewCell.h"

@import BVSDK;

@interface ViewController () <UITableViewDataSource>

@property (weak, nonatomic) IBOutlet BVReviewsTableView *reviewsTableView;

@property NSArray<BVReview *> *reviews;

@end


@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    
    self.reviews = [NSArray array];
    
    self.reviewsTableView.dataSource = self;
    self.reviewsTableView.estimatedRowHeight = 44;
    self.reviewsTableView.rowHeight = UITableViewAutomaticDimension;
    [self.reviewsTableView registerNib:[UINib nibWithNibName:@"MyReviewTableViewCell" bundle:nil] forCellReuseIdentifier:@"MyReviewTableViewCell"];
    
    BVReviewsRequest* request = [[BVReviewsRequest alloc] initWithProductId:@"test1" limit:20 offset:0];
    [self.reviewsTableView load:request success:^(BVReviewsResponse * _Nonnull response) {
        self.reviews = response.results;
        [self.reviewsTableView reloadData];
    } failure:^(NSArray<NSError *> * _Nonnull errors) {
        NSLog(@"Error loading reviews");
    }];

}


#pragma mark UITableViewDatasource

-(NSString *)tableView:(UITableView *)tableView titleForHeaderInSection:(NSInteger)section{
    return @"Review Responses";
}

-(NSInteger)tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger)section{
    return [self.reviews count];
}

- (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(NSIndexPath *)indexPath{
    
    MyReviewTableViewCell *cell = [tableView dequeueReusableCellWithIdentifier:@"MyReviewTableViewCell"];

    cell.review = [self.reviews objectAtIndex:indexPath.row];

    return cell;
}


@end

File: Examples/Conversations/Swift/ConversationsExample/ViewController.swift
//
//  ViewController.swift
//  Conversations
//
//  Copyright © 2016 Bazaarvoice. All rights reserved.
//

import UIKit
import BVSDK


class ViewController: UIViewController, UITableViewDataSource {
    
    @IBOutlet weak var reviewsTableView : BVReviewsTableView!
    var reviews : [BVReview] = []

    override func viewDidLoad() {
        super.viewDidLoad()
        
        reviewsTableView.dataSource = self
        reviewsTableView.estimatedRowHeight = 44
        reviewsTableView.rowHeight = UITableViewAutomaticDimension
        reviewsTableView.registerNib(UINib(nibName: "MyReviewTableViewCell", bundle: nil), forCellReuseIdentifier: "MyReviewTableViewCell")
        
        let reviewsRequest = BVReviewsRequest(productId: "test1", limit: 20, offset: 0)
        reviewsRequest.addReviewSort(.SubmissionTime, order: .Descending)
        
        reviewsTableView.load(reviewsRequest, success: { (response) in
            
            self.reviews = response.results
            self.reviewsTableView.reloadData()
            
            }) { (error) in
                
            print(error)
                
        }
        
    }
    
    // MARK: UITableViewDatasource
    
    func tableView(tableView: UITableView, titleForHeaderInSection section: Int) -> String? {
        return "Review Responses"
    }
    
    func tableView(tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
        return reviews.count
    }
    
    func tableView(tableView: UITableView, cellForRowAtIndexPath indexPath: NSIndexPath) -> UITableViewCell {
        let tableCell = tableView.dequeueReusableCellWithIdentifier("MyReviewTableViewCell") as! MyReviewTableViewCell
        
        tableCell.review = reviews[indexPath.row]
        
        return tableCell
    }

}
